test/unit/queue/src/CollectiveQueue.cpp

```diff
@@ -0,0 +1,174 @@
+            results[threadId] = threadId;
+        };
+
+        // The kernel will be performed collectively.
+        // OpenMP will distribute the work between the threads from the parallel region
+        alpaka::kernel::exec<Acc>(
+               queue,
+               workDiv,
+               kernel);
+
+        alpaka::wait::wait(queue);
+    }
+
+    for(size_t i = 0; i < results.size(); ++i)
+    {
+        REQUIRE(static_cast<int>(i) == results.at(i));
+    }
+}
+};
+
+//-----------------------------------------------------------------------------
+//! test that only one thread performs the copy operation
+struct TestCollectiveMemcpy
+{
+
+void operator()()
+{
+    // Define the index domain
+    using Dim = alpaka::dim::DimInt<1>;
+    using Idx = size_t;
+
+    // Define the accelerator
+    using Acc = alpaka::acc::AccCpuOmp2Blocks<Dim, Idx>;
+    using Dev = alpaka::dev::Dev<Acc>;
+
+    using Queue = alpaka::queue::QueueCpuOmp2Collective;
+    using Pltf = alpaka::pltf::Pltf<Dev>;
+
+    auto dev = alpaka::pltf::getDevByIdx<Pltf>(0u);
+    Queue queue(dev);
+
+    std::vector<int> results(4, -1);
+
+    // Define the work division
+    using Vec = alpaka::vec::Vec<Dim, Idx>;
+    Vec const elementsPerThread(Vec::all(static_cast<Idx>(1)));
+    Vec const threadsPerBlock(Vec::all(static_cast<Idx>(1)));
+    Vec const blocksPerGrid(results.size());
+
+    using WorkDiv = alpaka::workdiv::WorkDivMembers<Dim, Idx>;
+    WorkDiv const workDiv(
+        blocksPerGrid,
+        threadsPerBlock,
+        elementsPerThread);
+
+    #pragma omp parallel num_threads(static_cast<int>(results.size()))
+    {
+        int threadId = omp_get_thread_num();
+
+        using View = alpaka::mem::view::ViewPlainPtr<Dev, int, Dim, Idx>;
+
+        View dst(
+            results.data() + threadId,
+            dev,
+            Vec(1lu),
+            Vec(sizeof(int)));
+
+        View src(
+            &threadId,
+            dev,
+            Vec(1lu),
+            Vec(sizeof(int)));
+
+        // avoid that the first thread is executing the copy (can not be guaranteed)
+        size_t sleep_ms = (results.size() - static_cast<uint32_t>(threadId)) * 100u;
+        std::this_thread::sleep_for(std::chrono::milliseconds(sleep_ms));
+
+        // only one thread will perform this memcpy
+        alpaka::mem::view::copy(queue, dst, src, Vec(1lu));
+
+        alpaka::wait::wait(queue);
+    }
+
+    uint32_t numFlippedValues = 0u;
+    uint32_t numNonIntitialValues = 0u;
+    for(size_t i = 0; i < results.size(); ++i)
+    {
+        if(static_cast<int>(i) == results.at(i))
+            numFlippedValues++;
+        if(results.at(i) != -1)
+            numNonIntitialValues++;
+    }
+    // only one thread is allowed to flip the value
+    REQUIRE(numFlippedValues == 1u);
+    // only one value is allowed to differ from the initial value
+    REQUIRE(numNonIntitialValues == 1u);
+}
+};
+
+TEST_CASE( "queueCollective", "[queue]")
+{
+    TestCollective{}();
```

Comment:
Do we need the separate class at all? It is only used in the templatized tests as workaround. We should be able to move the code directly in here.